python3/image.bzl

```diff
@@ -0,0 +1,92 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A rule for creating a Python container image.
+
+The signature of this rule is compatible with py_binary.
+"""
+
+load(
+    "//lang:image.bzl",
+    "dep_layer",
+    "app_layer",
+)
+load(
+    "//container:container.bzl",
+    "container_pull",
+    _repositories = "repositories",
+)
+
+# Load the resolved digests.
+load(":python3.bzl", "DIGESTS")
+
+def repositories():
+  # Call the core "repositories" function to reduce boilerplate.
+  # This is idempotent if folks call it themselves.
+  _repositories()
+
+  excludes = native.existing_rules().keys()
+  if "py3_image_base" not in excludes:
+    container_pull(
+      name = "py3_image_base",
+      registry = "gcr.io",
+      repository = "distroless/python3",
+      digest = DIGESTS["latest"],
+    )
+  if "py_debug_image_base" not in excludes:
+    container_pull(
+      name = "py_debug_image_base",
+      registry = "gcr.io",
+      repository = "distroless/python3",
+      digest = DIGESTS["debug"],
+    )
+
+DEFAULT_BASE = select({
+    "@io_bazel_rules_docker//:fastbuild": "@py3_image_base//image",
+    "@io_bazel_rules_docker//:debug": "@py3_debug_image_base//image",
+    "@io_bazel_rules_docker//:optimized": "@py3_image_base//image",
+    "//conditions:default": "@py3_image_base//image",
+})
+
+def py3_image(name, base=None, deps=[], layers=[], **kwargs):
+  """Constructs a container image wrapping a py_binary target.
+
+  Args:
+    layers: Augments "deps" with dependencies that should be put into
+           their own layers.
+    **kwargs: See py_binary.
+  """
+  binary_name = name + ".binary"
+
+  # TODO(mattmoor): Consider using par_binary instead, so that
+  # a single target can be used for all three.
+  native.py_binary(name=binary_name, deps=deps + layers, **kwargs)
+
+  # TODO(mattmoor): Consider what the right way to switch between
+  # Python 2/3 support might be.  Perhaps just overriding `base`,
+  # but perhaps we can be smarter about selecting a py2 vs. py3
+  # distroless base?
```

Comment:
Seems like we're making a call here :)

Can you drop it here and the source?

Reply:
Done.

Though I'm still on the fence about the approach here: we could logically extend it toward treating `py_binary` and `py_library` as python 2-only, and then adding `py3_binary` and `py3_library` rules to `rules_python` and so forth.  They are two different languages in some senses, though I don't know if I want that precedent here.

Reply:
I will leave that to the Python gods to decide.  I just play one on GitHub :)

@duggelz FYI